product_assortment/security/product_assortment_security.xml

```diff
@@ -0,0 +1,23 @@
+<?xml version="1.0" encoding="utf-8" ?>
+<odoo noupdate="1">
+    <record id="group_product_assortment_manager" model="res.groups">
+        <field name="name">Product Assortment Manager</field>
+        <field name="comment">Access to all product assortment</field>
+    </record>
+    <record id="base.ir_filters_employee_rule" model="ir.rule">
+        <field
+            name="domain_force"
+        >[('user_id','in',[False,user.id]), ('is_assortment', '=', False)]</field>
```

Comment:
@sergio-teruel I think this is not very resilient, as other modules modifying the same rule will make this fail. What about overriding the `ir.rule` `_compute_domain` to inject the extra domain for this one? Example: https://github.com/OCA/sale-workflow/blob/16.0/sales_team_security/models/ir_rule.py#L23